modules/core/src/main/java/org/apache/synapse/message/store/impl/rabbitmq/RabbitMQConsumer.java

```diff
@@ -34,255 +34,220 @@
-		isReceiveError = false;
-		isInitialized = true;
-	}
-
-	public MessageContext receive() {
-		if (!checkConnection()) {
-			if (!reconnect()) {
-				if (logger.isDebugEnabled()) {
-					logger.debug(getId()
-					             + " cannot receive message from store. Can not reconnect.");
-				}
-				return null;
-			} else {
-				logger.info(getId() + " reconnected to store.");
-				isReceiveError = false;
-			}
-		}
-		//setting channel
-		if (channel != null) {
-			if (!channel.isOpen()) {
-				if (!setChannel()) {
-					logger.info(getId() + " unable to create the channel.");
-					return null;
-				}
-			}
-		} else {
-			if (!setChannel()) {
-				logger.info(getId() + " unable to create the channel.");
-				return null;
-			}
-		}
-		//receive messages
-		try {
 
-			GetResponse delivery = null;
-			delivery = channel.basicGet(queueName, false);
-
-			if (delivery != null) {
-				//deserilizing message
-				StorableMessage storableMessage = null;
-				ByteArrayInputStream bis = new ByteArrayInputStream(delivery.getBody());
-				ObjectInput in = new ObjectInputStream(bis);
-				try {
-					storableMessage = (StorableMessage) in.readObject();
-				} catch (ClassNotFoundException e) {
-					logger.error(getId() + "unable to read the stored message" + e);
-					channel.basicAck(delivery.getEnvelope().getDeliveryTag(), false);
-				}
+    private static final Log log = LogFactory.getLog(RabbitMQConsumer.class.getName());
+
+    private Connection connection;
+    private Channel channel;
+    private RabbitMQStore store;
+    private String queueName;
+    private String idString;
+    private CachedMessage cachedMessage; // Holds the last message read from the message store
+
+    public RabbitMQConsumer(RabbitMQStore store) {
+        if (store == null) {
+            log.error("Cannot initialize.");
+            return;
+        }
+        this.store = store;
+        cachedMessage = new CachedMessage();
+    }
 
-				bis.close();
-				in.close();
-				org.apache.axis2.context.MessageContext axis2Mc = store.newAxis2Mc();
-				MessageContext synapseMc = store.newSynapseMc(axis2Mc);
-				synapseMc =
-						MessageConverter.toMessageContext(storableMessage, axis2Mc, synapseMc);
-				updateCache(delivery, synapseMc, null, false);
-				if (logger.isDebugEnabled()) {
-					logger.debug(getId() + " Received MessageId:" +
-					             delivery.getProps().getMessageId());
-				}
-				return synapseMc;
-			}
-		} catch (ShutdownSignalException sse) {
-			logger.error(getId() + " connection error when receiving messages" + sse);
-		} catch (IOException ioe) {
-			logger.error(getId() + " connection error when receiving messages" + ioe);
-		}
-		return null;
-	}
+    /**
+     * Get a single message from the queue and deserialize for set into the message context
+     *
+     * @return the {@link MessageContext} with received message
+     */
+    @Override
+    public MessageContext receive() {
+        if (isAlive()) {
```

Comment:
This does not check if message processor is live, rather it checks if connection is live. Shall we change the method name?

Reply:
The `isAlive` method suppose to check the availability of connectivity with the message store. The processor must be deactivated if the `isAlive` returns false. That method inherited from the `MessageConsumer`.